From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 9 Sep 2017 18:17:25 +0200
Subject: [PATCH] {For LLVMLinux} compiler-gcc.h: do not use __optimize__ with
 clang

clang does not know about this attribute:

    arch/x86/mm/mem_encrypt.c:526:13: error: unknown attribute
    '__optimize__' ignored [-Werror,-Wunknown-attributes]
    void __init __nostackprotector sme_enable(struct boot_params *bp)
                ^
    include/linux/compiler-gcc.h:170:43: note: expanded from macro
    '__nostackprotector'
    #define __nostackprotector
    __attribute__((__optimize__("no-stack-protector")))
                   ^
---
 include/linux/compiler-gcc.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
index 3ec0a7907d06..25206772e47e 100644
--- a/include/linux/compiler-gcc.h
+++ b/include/linux/compiler-gcc.h
@@ -167,8 +167,10 @@
 #if GCC_VERSION >= 40100
 # define __compiletime_object_size(obj) __builtin_object_size(obj, 0)
 
+#ifndef __clang__
 #define __nostackprotector	__attribute__((__optimize__("no-stack-protector")))
 #endif
+#endif
 
 #if GCC_VERSION >= 40300
 /* Mark functions as cold. gcc will assume any path leading to a call
-- 
